tools/ci/canary-runner/src/run.rs

```diff
@@ -44,6 +46,60 @@ pub struct RunOpt {
 }
 
 pub async fn run(opt: RunOpt) -> Result<()> {
+    let start_time = SystemTime::now();
+    let config = aws_config::load_from_env().await;
+    let result = run_canary(opt, &config).await;
+
+    let mut metrics = vec![
+        (
+            "canary-success",
+            if result.is_ok() { 1.0 } else { 0.0 },
+            StandardUnit::Count,
+        ),
+        (
+            "canary-failure",
+            if result.is_ok() { 0.0 } else { 1.0 },
+            StandardUnit::Count,
+        ),
+        (
+            "canary-total-time",
+            start_time.elapsed().expect("time in range").as_secs_f64(),
+            StandardUnit::Seconds,
+        ),
+    ];
+    if let Ok(invoke_time) = result {
+        metrics.push((
+            "canary-invoke-time",
+            invoke_time.as_secs_f64(),
+            StandardUnit::Seconds,
+        ));
+    }
+
+    let cloudwatch_client = cloudwatch::Client::new(&config);
+    let mut request_builder = cloudwatch_client
+        .put_metric_data()
+        .namespace("aws-sdk-rust-canary");
+    for metric in metrics {
+        request_builder = request_builder.metric_data(
+            cloudwatch::model::MetricDatum::builder()
+                .metric_name(metric.0)
+                .value(metric.1)
+                .timestamp(SystemTime::now().into())
+                .unit(metric.2)
+                .build(),
+        );
+    }
+
+    info!("Emitting metrics...");
+    request_builder
+        .send()
+        .await
+        .context("failed to emit metrics")?;
```

Comment:
is there some sort of fallback where this can fail even if we somehow bricked the cloudwatch client?

Reply:
My thought was that once we have the canary running on a regular cadence, we can set up an alarm that goes off if it hasn't logged a metric in the last 25 or so hours.